usefulUtilities/statusIndicator/webApp/app/statusIndicator_webApp.js

```diff
@@ -0,0 +1,325 @@
+//
+// statusIndicator_webApp.js
+//
+// NodeJS Web App for Status Indicator App
+// Created by Robin Wilson on 2019-04-02
+// Copyright High Fidelity 2019
+//
+// Licensed under the Apache 2.0 License
+// See accompanying license file or http://apache.org/
+
+var http = require('http');
+var url = require('url');
+var dbInfo = require('./dbInfo.json');
+
+
+// Heartbeat updates setTimeout and updates employee data in database
+var users = {}; // { username: { timer: <timer object>} }
+function heartbeat(queryParamObject, response) {
+    var username = queryParamObject.username;
+    // handle heartbeat
+    if (!users[username]) {
+        // new heartbeat for username
+        users[username] = {};
+    } else {
+        // stop the old heartbeat timer
+        clearTimeout(users[username].timer);
+    }
+    users[username].timer = startHeartbeatTimer(username, response);
+
+    if (username) {
+        updateEmployee(queryParamObject, response);
+    } else {
+        console.error("There was no username or primary key! Can't update or create employee!");
+        return;
+    }
+}
+
+
+// If timer runs out, will set user to offline
+var HEARTBEAT_INTERVAL_MS = 10000;
```

Comment:
it's not _really_ an interval, so this might be a little confusing. have you considered renaming this to something like `HEARTBEAT_TIMEOUT_MS` ?